### ISSUE TITLE
Explain 'sudo -E' someplace accessible

### ISSUE THREAD

**Issue by yuvipanda** [yuvipanda/the-littlest-jupyterhub#11]
It's important, but need to figure out where to explain it without complicating things

**Comment by choldgraf** [comment]
is the basic point that this allows you to run sudo and keep the current PATH available?

**Comment by consideRatio** [comment]
I looked and made some inline comments on the z2jh ci scripts today.

sudo -E can be written out more explicitly with sudo --preserve-env, that is a good start perhaps?
One complication I've learned about also, is that the PATH env variable is a special variable that won't be preserved even though you want it to be for security reasons.

**Comment by yuvipanda** [comment]
Yep, it keeps PATH available *because* we make it be available for users in the jupyterhub-admins group with this piece of config: https://github.com/jupyterhub/the-littlest-jupyterhub/blob/master/tljh/installer.py#L190

**Comment by kannes** [comment]
I would also recommend "-H, --set-home" so that users do not get warnings like

> The directory '/home/jupyter-adminuser/.cache/pip/http' or its parent directory is not owned by the current user and the cache has been disabled. Please check the permissions and owner of that directory. If executing pip with sudo, you may want sudo's -H flag.
> The directory '/home/jupyter-adminuser/.cache/pip' or its parent directory is not owned by the current user and caching wheels has been disabled. check the permissions and owner of that directory. If executing pip with sudo, you may want sudo's -H flag.

**Comment by fm75** [comment]
I have a PR in that adds the -E as part of solving a problem where the installation is behind a proxy.
